pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -54,8 +54,9 @@ const (
 	enforceHostNetworkSettingName  string = "ROOK_ENFORCE_HOST_NETWORK"
 	enforceHostNetworkDefaultValue string = "false"
 
-	revisionHistoryLimitSettingName  string = "ROOK_REVISION_HISTORY_LIMIT"
-	revisionHistoryLimitDefaultValue string = ""
+	revisionHistoryLimitSettingName string = "ROOK_REVISION_HISTORY_LIMIT"
+	revisionHistoryLimitDefaultStr  string = ""
+	revisionHistoryLimitDefault     int32  = int32(10)
```

Comment:
Kubernetes already keeps its own default, and if users are able to change the kube default in the future, this could mess that up for them. Let's avoid Rook having its own default.

Reply:
@BlaineEXE  wrote:
> Kubernetes already keeps its own default, and if users are able to change the kube default in the future, this could mess that up for them. Let's avoid Rook having its own default.

I did try to get along without adding a default value, but somehow, I did not manage to get the code compiling and passing tests without one.

Reply:
Can this be set to nil instead of a specific value?

Reply:
@BlaineEXE , I agree completely that we should not be setting an explicit default value but instead rely on the implicit kubernetes default!

But I need to set an `*int32` variable. If I set it to `nil`, this creates panics (nil pointer deref).  Any suggestion?

Reply:
I don't see any strong reason to need to handle pointer dereference outside of the functions that parse the values. Kubernetes's API types are `*int32` which will accept a nil pointer or a non-nil pointer without issues.

Reply:
Since we are working to remove the hard coded Rook default, let's remove this line.

Reply:
done. removed the hard-coded default.